minicluster/src/main/java/alluxio/master/AbstractLocalAlluxioCluster.java

```diff
@@ -468,6 +469,16 @@ protected void setHostname() {
   protected void setAlluxioHome() throws IOException {
     mHome =
         File.createTempFile("Alluxio", "U" + System.currentTimeMillis()).getAbsolutePath();
+    // Make a copy of mHome.
+    final String home = mHome;
+    Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
```

Comment:
Q: is it considered as a good practice to invoke `addShutdownHook`?
also, is it possible to leverage junit temporary dir ?